aloe/registry.py

```diff
@@ -449,3 +454,111 @@ def decorator(self, function, **kwargs):
 around = CallbackDecorator(CALLBACK_REGISTRY, 'around')
 before = CallbackDecorator(CALLBACK_REGISTRY, 'before')
 # pylint:enable=invalid-name
+
+
+class ExtendedStep(object):
+    """
+    An extended step decorator that defines placeholders and allows to easily
+    assign multiple sentences to a single function.
+    Useful for sentences that capture a string which is enclosed either in
+    single or double quotes
+    """
+    CLOSURE = '{%s}'
+
+    def __init__(self):
+        self.single_expression_placeholders = {
+            'NUMBER': r'(-?\d+(?:\.\d*)?)',
+            'NON_CAPTURING_STRING': r'|'.join((r'"[^"]*"', r"'[^']*'")),
+            'NON_CAPTURING_NUMBER': r'-?\d+(?:\.\d*)?',
+        }
+        self.multi_expression_placeholders = {
```

Comment:
If you are wrapping everything with regexes, there's no need for this `multi` thing - you can define a single regex to do this. (Or if you can't, just wrap `single` into `multi` instead.)